sql-plugin/src/main/scala/com/nvidia/spark/rapids/BatchWithPartitionData.scala

```diff
@@ -0,0 +1,468 @@
+   */
+  private def splitPartitionDataIntoGroups(
+      partitionRowData: Array[PartitionRowData],
+      partSchema: StructType,
+      maxSizeOfBatch: Long): Array[Array[PartitionRowData]] = {
+    val allowedMaxSizeOfBatch = Math.min(CUDF_COLUMN_SIZE_LIMIT, maxSizeOfBatch)
+    val resultBatches = ArrayBuffer[Array[PartitionRowData]]()
+    val currentBatch = ArrayBuffer[PartitionRowData]()
+    val sizeOfBatch = Array.fill(partSchema.length)(0L)
+    var partIndex = 0
+    var splitOccurred = false
+    var rowsInPartition = 0
+    while (partIndex < partitionRowData.length) {
+      // If there was no split, get next partition else process remaining rows in current partition.
+      if (!splitOccurred) {
+        rowsInPartition = partitionRowData(partIndex).rowNum
+      }
+      val valuesInPartition = partitionRowData(partIndex).rowValue
+      // Calculate the maximum number of rows that can fit in current batch.
+      val maxRows = calculateMaxRows(valuesInPartition, partSchema, sizeOfBatch,
+        allowedMaxSizeOfBatch)
+      // Splitting occurs if for any column, maximum rows we can fit is less than rows in partition.
+      splitOccurred = maxRows.min < rowsInPartition
+      if (splitOccurred) {
+        currentBatch.append(PartitionRowData(valuesInPartition, maxRows.min))
+        resultBatches.append(currentBatch.toArray)
+        currentBatch.clear()
+        java.util.Arrays.fill(sizeOfBatch, 0)
+        rowsInPartition -= maxRows.min
+      } else {
+        // If there was no split, all rows can fit in current batch.
+        currentBatch.append(PartitionRowData(valuesInPartition, rowsInPartition))
+        val partitionSizes = calculatePartitionSizes(rowsInPartition, valuesInPartition, partSchema)
+        sizeOfBatch.indices.foreach(i => sizeOfBatch(i) += partitionSizes(i))
+        // Process next partition
+        partIndex += 1
+      }
+    }
+    // Add the last remaining batch to resultBatches
+    if (currentBatch.nonEmpty) {
+      resultBatches.append(currentBatch.toArray)
+    }
+    resultBatches.toArray
+  }
+
+  /**
+   * Calculates the partition size for each column as 'size of single value * number of rows'
+   */
+  private def calculatePartitionSizes(rowNum: Int, values: InternalRow,
+      partSchema: StructType): Seq[Long] = {
+    partSchema.zipWithIndex.map {
+      case (field, colIndex) if field.dataType == StringType =>
+        // Assumption: Columns of other data type would not have fit already. Do nothing.
+        val singleValue = values.getString(colIndex)
+        val sizeOfSingleValue = singleValue.getBytes("UTF-8").length.toLong
+        rowNum * sizeOfSingleValue
+      case _ => 0L
+     }
+  }
+
+  /**
+   * For each column, calculate the maximum number of rows that can fit into a batch
+   * without exceeding limit.
+   */
+  private def calculateMaxRows(valuesInPartition: InternalRow, partSchema: StructType,
+      sizeOfBatch: Array[Long], maxSizeOfBatch: Long): Seq[Int] = {
```

Comment:
This should perform the `.min` to simply return the maximum number of rows that can fit, as that more closely matches the name of the method and what the callers want anyway.

Reply:
Replaced with min